tests/spec/s-object.js

```diff
@@ -365,4 +379,48 @@ describe('Object', function () {
             expect(obj instanceof Object).toBe(false);
         });
     });
+
+    xdescribeGOPN('.getOwnPropertyNames()', function () {
+        // ES6 Object.getOwnPropertyNames does not require these throws
+        xit('primitives should throw `TypeError`', function () {
+            expect(function () {
+                Object.getOwnPropertyNames();
+            }).toThrow();
+            expect(function () {
+                Object.getOwnPropertyNames(undefined);
+            }).toThrow();
+            expect(function () {
+                Object.getOwnPropertyNames(null);
+            }).toThrow();
+            expect(function () {
+                Object.getOwnPropertyNames(1);
+            }).toThrow();
+            expect(function () {
+                Object.getOwnPropertyNames(true);
+            }).toThrow();
+            expect(function () {
+                Object.getOwnPropertyNames('abc');
+            }).toThrow();
+        });
+
+        itHasMap('works on Map', function () {
+          expect(Object.getOwnPropertyNames(new Map([[1, 2], [3, 4]]))).toEqual([]);
+        });
+
+        itHasSet('works on Set', function () {
+          expect(Object.getOwnPropertyNames(new Set([1, 2, 3, 4]))).toEqual([]);
+        });
+
+        itHasArrayBuffer('works on ArrayBuffer', function () {
+          expect(Object.getOwnPropertyNames(new ArrayBuffer(4))).toEqual([]);
+        });
+
+        itHasUint8Array('works on Uint8Array', function () {
+          expect(Object.getOwnPropertyNames(new Uint8Array(4))).toEqual(['0', '1', '2', '3']);
```

Comment:
That'd be a Map and Set test, not a getOwnPropertyNames test :-) thanks for reporting that, I'll see if I can fix it.